dist/tools/kconfiglib/examples/select_specific_implementation_from_module/README.md

```diff
@@ -0,0 +1,74 @@
+# Select a specific choice implementation from a module
+
+This is a design pattern when one must select a specific implementation from
+a module. This is specifically for modules that are not defined in boards or
+applications.
```

Comment:
> This is specifically for modules that are not defined in boards or
> applications.

Why is this?

Reply:
It covers it in the `### What are choice overrides only allowed in boards and applications?` section, I didn't want to repeat it but maybe it would make sense to link.

Reply:
I guess the answer would be, because you can just use a choice override.